### ISSUE TITLE
Issues with focus indicator on our `Link` component

### ISSUE THREAD

**Issue by Febakke** [digdir/designsystem#4042]
**WIP**

We have gotten feedback that our focus indicator on links cause some issues. When combining with icon/logo it can make the text not visible. It also might be a little "unstable" meaning that in some cases it will add the background color but not change the text color. When we have seen this before it might be an issue with how it has been set up by the developer or its a problem with our components. Either way, we should look into a version that is "stable". By stable i Mean that it dont wipe out all text if something is not set up correctly. 

##

**Comment by Barsnes** [comment]
We could use [`color-mix`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix) to automatically generate a contrasting text color

**Comment by mrosvik** [comment]
We have decided to change the focus style for links. The reason for this is that the previous focus style, which used a background fill, was vulnerable to coding errors where the text color was not updated, making the link unreadable. We have now aligned the link focus style with the one used for the rest of our components.

This is updated in Figma. 

**Old:** 
<img width="726" height="501" alt="Image" src="https://github.com/user-attachments/assets/38eed7e4-fe9c-40d5-8f86-9cbcde8096e9" />


**New:** 
<img width="688" height="475" alt="Image" src="https://github.com/user-attachments/assets/dd3cdf94-581c-4dfc-9e1f-5d59525af714" />
